packages/fcl/src/current-user/index.js

```diff
@@ -0,0 +1,268 @@
+}
+
+const currentUserLogic = async ctx => {
+  saveData(ctx, JSON.parse(DATA))
+
+  __loop: while (1) {
+    const letter = await ctx.receive()
+
+    try {
+      await HANDLERS[letter.tag](ctx, letter, letter.data || {})
+    } catch (error) {
+      console.error("Current User Error", letter, error)
+    } finally {
+      continue __loop
+    }
+  }
+}
+
+const identity = v => v
+const spawnCurrentUser = () => spawn(currentUserLogic, NAME)
+
+function renderFrame({handshake, scope, nonce, l6n}) {
+  if (document.getElementById(FRAME_ID)) return
+
+  const src = [
+    handshake,
+    [
+      `l6n=${encodeURIComponent(l6n)}`,
+      `nonce=${encodeURIComponent(nonce)}`,
+      scope && `scope=${scope.split(" ").join("+")}`,
+    ]
+      .filter(Boolean)
+      .join("&"),
+  ].join("?")
+
+  const $frame = document.createElement("iframe")
+  $frame.src = src
+  $frame.id = FRAME_ID
+  $frame.style.height = "500px"
+  $frame.style.maxHeight = "90vh"
+  $frame.style.width = "400px"
+  $frame.style.maxWidth = "90vw"
+  $frame.style.display = "block"
+  $frame.style.background = "#fff"
+  $frame.style.position = "fixed"
+  $frame.style.top = "5vh"
+  $frame.style.right = "calc(50vw)"
+  $frame.style.transform = "translateX(50%)"
+  $frame.style.boxShadow = "0 4px 8px -4px black"
+  $frame.frameBorder = "0"
+  document.body.append($frame)
+}
+
+const compositeId = data => {
+  const addr = ((data || {}).provider || {}).addr
+  const pid = ((data || {}).provider || {}).pid
+  if (addr == null || pid == null) return null
+  return `${addr}/${pid}`
+}
+
+async function authenticate() {
+  return new Promise(async resolve => {
+    spawnCurrentUser()
+
+    const user = await snapshot()
+    if (user.loggedIn) return resolve(user)
+
+    const handshake = await config().get("challenge.handshake")
+    const scope = await config().get("challenge.scope")
+    const nonce = "asdf"
+    const l6n = window.location.origin
+
+    renderFrame({handshake, scope, nonce, l6n})
+
+    window.addEventListener("message", async ({data, origin}) => {
+      if (data.type !== CHALLENGE_RESPONSE_EVENT) return
+      if (document.getElementById(FRAME_ID)) {
+        document.getElementById(FRAME_ID).remove()
+      }
+
+      const user = await fetch(`${data.hks}?code=${data.code}`, {
+        method: "GET",
+        headers: {
+          "Content-Type": "application/json",
+        },
+      }).then(d => d.json())
+
+      send(NAME, SET_CURRENT_USER, {
+        ...user,
+        cid: compositeId(user),
+        loggedIn: true,
+        verified: true,
+      })
+      resolve(await snapshot())
+    })
+  })
+}
+
+function unauthenticate() {
+  spawnCurrentUser()
+  send(NAME, DEL_CURRENT_USER)
+}
+
+async function authorization(account) {
+  spawnCurrentUser()
+  await authenticate()
+
+  const user = await snapshot()
+  let sequenceNum
+  if (account.role.proposer) {
+    const acct = await info()
+    // TODO: There will be an update to the getAccount that will
```

Comment:
Im not sure if I understand this todo.. are you talking about the format of a key through the sdk or is this referring to something else?

Reply:
send get account is currently returning this shape: 
```javascript
{
  account: {
    keys: [
     { index, sequenceNumber, publicKey }
   ]
  }
}
```
You will be making a change to send soon that will make it return this shape
```javascript
{
  account: {
    keys: [
     { keyId, sequenceNum, publicKey }
   ]
  }
}
```
So that it matches the naming in the interaction...